pkg/api/types.go

```diff
@@ -498,6 +498,33 @@ type EnvVar struct {
 	Name string `json:"name"`
 	// Optional: defaults to "".
 	Value string `json:"value,omitempty"`
+	// Optional: specify a source for the value of the EnvVar.
+	*EnvVarSource `json:"source,omitempty"`
+}
+
+// EnvVarSource represents a source for the value of an EnvVar.
+// Only one of its members may be specified.
+type EnvVarSource struct {
+	// ObjectInfo represents a downward API request for a field of the object.
+	//
+	// Valid fields of this reference are:
+	//
+	// 1. APIVersion
+	// 2. FieldPath
+	//
+	// No other fields may be specified.
```

Comment:
This feels like an abuse of an existing type.  Why not just define your own type?